src/test/java/com/wikia/webdriver/common/logging/Log.java

```diff
@@ -0,0 +1,338 @@
+    log(command, description, success, false);
+  }
+
+  public static void log(String command, Throwable e, boolean success) {
+    log(command, e.getMessage(), success, false);
+  }
+
+  public static void log(String command, String descriptionOnSuccess, String descriptionOnFail,
+                         boolean success) {
+    String description = descriptionOnFail;
+    if (success) {
+      description = descriptionOnSuccess;
+    }
+    log(command, description, success, false);
+  }
+
+  /**
+   * LogData an action that is not user facing. LogData file reader can hide these actions to
+   * increase test readability
+   */
+  public static void logOnLowLevel(String command, String description, boolean success) {
+    log(command, description, success, true);
+  }
+
+  private static void log(String command, String description, boolean isSuccess,
+                          boolean ifLowLevel) {
+    LOGS_RESULTS.add(isSuccess);
+    String escapedDescription = escapeHtml(description);
+
+    List<LogData> logTypeList = new ArrayList<>();
+    logTypeList.add(isSuccess ? LogLevel.OK : LogLevel.ERROR);
+
+    if (ifLowLevel) {
+      logTypeList.add(LogLevel.DEBUG);
+    }
+    VelocityWrapper.fillLogRow(logTypeList, command, escapedDescription);
+    logJSError();
+  }
+
+  public static void ok(String command, String description) {
+    log(command, description, true);
+  }
+
+  public static void logError(String command, Throwable throwable) {
+    log(command, escapeHtml(throwable.toString()), false, DriverProvider.getActiveDriver());
+    stacktrace(throwable);
+  }
+
+  public static void warning(String command, Exception exception) {
+    warning(command, exception.getMessage());
+  }
+
+  /**
+   * This method will log warning to log file (line in yellow color)
+   */
+  public static void warning(String command, String description) {
+    VelocityWrapper.fillLogRow(Arrays.asList(LogLevel.WARNING), command, description);
+  }
+
+  /**
+   * This method will log info to log file (line in blue color)
+   */
+  public static void info(String description) {
+    VelocityWrapper.fillLogRow(Arrays.asList(LogLevel.INFO), "INFO", description);
+  }
+
+  /**
+   * This method will log info to log file (line in blue color)
+   */
+  public static void info(String command, String description) {
+    VelocityWrapper.fillLogRow(Arrays.asList(LogLevel.INFO), command, description);
+  }
+
+  /**
+   * This method will log info to log file (line in blue color)
+   */
+  public static void info(String description, Throwable e) {
+    String finalDescription = description + " : " + e.getMessage();
+    VelocityWrapper.fillLogRow(Arrays.asList(LogLevel.INFO), "INFO", finalDescription);
+  }
+
+  public static void image(String command, File image, boolean success) {
+    byte[] bytes = new byte[0];
+    try {
+      bytes = new Base64().encode(FileUtils.readFileToByteArray(image));
+    } catch (IOException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)